crates/core/types/transaction.rs

```diff
@@ -805,6 +851,41 @@ mod serde_impl {
             struct_serializer.end()
         }
     }
+
+    /// Unsigned Transaction struct generic to all types
```

Comment:
Again, I think I would move this to rpc crate to discourage to use this outside of rpc context. Thoughts?

Reply:
It is also used by the evm crate to create access list and estimate gas. In both of these cases not all of the transaction fields are required so we use the generic transaction instead of mapping it to a `Transaction`

Reply:
And we would run into cyclic dependencies if we were to import an rpc type into the evm crate

Reply:
Why can't we parse a `GenericTransaction` inside rpc crate, then convert it to a core `Transaction` and send that one to the `evm`? Otherwise we're leaking a rpc abstraction to the whole codebase.

Reply:
This way the evm will have to deal only with Transaction.

Reply:
We don't have all the fields needed for a `Transaction`

Reply:
Maybe we should reword the comment so that it is not confused with an rpc-specific struct